crates/core/src/connection.rs

```diff
@@ -190,6 +195,26 @@ impl Connection {
         self.transaction_with_behavior(TransactionBehavior::Deferred)
     }
 
+    pub(crate) async fn execute2<S, P>(&self, sql: S, params: P) -> Result<u64>
```

Comment:
I usually do `fn2` to signify that this is a new version which we plan to upgrade too. It makes it easier to dev in the meantime so we don't break old uses of the api. Since these are private these are easy things to clean up down the line. (aka we would remove the original and rename this one back to `execute`).